_posts/2021-12-15-dd_00004_logic_gates_2.md

```diff
@@ -0,0 +1,343 @@
+so-called **layout**. The layout describes how the individial transistors and
+the wires connecting them should be _layed out_ on the final chip. The layout is
+the final _blueprint_ for the manufacturing. We have to admit that the picture
+above does not show a real photo[^2] of a NAND gate. However, the structures are
+infact how they would look on the final chip, though without any colors.
+
+[^2]:
+    Taking a photo of nm-sized gates using a conventional camera is not
+    possible, as visible light has a much bigger wavelength compared to modern
+    gates sizes! However, we can use a _scanning electron microscope_ instead.
+    Check out this [video](https://www.youtube.com/watch?v=Fxv3JoS1uY8) if you
+    are interested.
+
+Physical design and layout in itself is so complex, that there are engineers who
+just work on optimizing the layout of a chip. Nowaday, with the advent of
+sophisticated EDA tools, this task is being automated more and more. However, a
+lot of reasearch and manual handwork still goes into optimizing the physical
+layout of _individual_ gates, in order to make them even smaller and faster.
+That's why the NAND gate layout above doesn't look at all like the schematic at
+first glance. However, the layout was simply optimized in such a way to consume
+the least amount of area (chip area = more money you have to pay the foundry),
+but the fundamental transistors and their connections are still there, so the
+layout it still implementing the NAND gate's schematic.
+
+## Summary
+
+If this post was too much information to take in, don't worry. In this blog
+series we will focus on higher level digital design, and the lowest we will ever
+go to is the level of basic logic gates. With this post we just wanted to show
+you how the gates are actually implemented in hardware, so you can at least say
+_"I have seen how logic gates are implemented on a chip"_, and that already
+would make us happy :)
+
+As we now have reached the end of this post, let us quickly summarize the key
+ideas we have learned:
+
+- **CMOS** stands for "Complementary Metal-Oxide Semiconductor"
+- In CMOS there are always **PMOS/NMOS** pairs
+- The PMOS transistors are part of the **pull-up network**, and the NMOS
+  transistors belong to the **pull-down network**
+- The CMOS **inverter** consists of one PMOS and one NMOS transistor
+- The CMOS **NAND** and **NOR** gates consist of two PMOS and two NMOS
+  transistors
+- We have learned about **series** and **parallel** connections
+- An **AND** gate is built using a NAND gate followed by an inverter
+  - Similar for the **OR** gate: NOR followed by inverter
+- We have learned what a **physical layout** is, and, as an example, seen how a
+  NAND gate looks in the layout
+
+We hope you liked this post, and as always thanks for reading! Follow us on
+[Twitter](https://twitter.com/bitsandvolts) to don't miss any of our future
+posts!
```

Comment:
Maybe a teaser for next post?

Reply:
Not sure yet what the next post will be about

Reply:
then it is alright